gapis/api/vulkan/api/queue.api

```diff
@@ -113,9 +113,16 @@ cmd VkResult vkQueueSubmit(
           wait_semaphores_all_valid.b = false
           vkErrorInvalidSemaphore(ws)
         } else {
-          LastBoundQueue.PendingCommands[len(LastBoundQueue.PendingCommands)]
-          = new!CommandReference(as!VkCommandBuffer(0), 0, cmd_vkNoCommand, 0,
-            Unsignal,    ws,    null, as!VkFence(0))
+          semaphore := Semaphores[ws]
+
+          if(semaphore.Submitted == 0) {
+            vkErrorSemaphoreNotSubmitted(ws)
+          } else {
+            LastBoundQueue.PendingCommands[len(LastBoundQueue.PendingCommands)]
+            = new!CommandReference(as!VkCommandBuffer(0), 0, cmd_vkNoCommand, 0,
+              Unsignal,    ws,    null, as!VkFence(0))
+            semaphore.Submitted = semaphore.Submitted - 1
```

Comment:
I am a bit worried we might be able to get in an underflow state, which would screw everything up.

Reply:
Thats true. I will change it to a signed variable.

Reply:
Not just a signed variable, I think we actually want to clamp it

Reply:
A negative submit count is invalid (this is just to handle invalid cases)
So I think the solution will be
```
semaphore.Submitted = switch semaphore.Submitted {
 case 0:
      0
  default:
      semaphore.Submitted - 1
}
```

Reply:
There we are already checking if it is 0 before. But we don't check in vkQueuePresentKHR though. I think if we add a check there too, it shouldn't be an issue.

Reply:
Sure that works for me.